docs/howto/mock_services.md

```diff
@@ -78,126 +78,166 @@ ZIO Test provides a framework for mocking your modules.
-  object multiArgs          extends Method[Service, (Int, Long), String]
-  object multiParamLists    extends Method[Service, (Int, Long), String]
-  object command            extends Method[Service, Int, Unit]
-  object overloaded {
-    object _1 extends Method[Service, Int, String]
-    object _2 extends Method[Service, Long, String]
-  }
+  def processEvent(event: AccountEvent) =
+    ZIO.accessM[AccountObserver](_.get.processEvent(event))
 }
-```
 
-> **Note:** we're using tuples to represent multiple argument methods, which follows with a limit to max 22 arguments, as is Scala itself limited.
+// test sources
+object AccountObserverMock {
 
-For overloaded methods we simply nest a list of numbered objects, each representing subsequent overloads.
+  sealed trait Tag[I, A] extends Method[AccountObserver, I, A] {
+    val mock: ZLayer[Has[Mock], Nothing, AccountObserver] =
+      AccountObserverMock.Mock
+  }
 
-Next, we create the mockable implementation of the service:
+  object ProcessEvent extends Tag[AccountEvent, Unit]
 
-```scala mdoc:silent
-implicit val mockableAccountObserver: Mockable[AccountObserver.Service] = (mock: Mock) =>
-  Has(new AccountObserver.Service {
-    def processEvent(event: AccountEvent): UIO[Unit] = mock(AccountObserver.Service.processEvent, event)
-  })
+  private val Mock: ZLayer[Has[Mock], Nothing, AccountObserver] =
+    ZLayer.fromService(mock =>
+      new AccountObserver.Service {
+        def processEvent(event: AccountEvent) = mock(ProcessEvent, event)
+      }
+    )
+}
 ```
 
-> **Note:** To make our mockable implementation automatically discovered, we need to place it inside `AccountObserver` module's companion object.
-
 ## Scrapping the boilerplate
 
-All of this machinery is repetitive and boring work, prone to simple mistakes. Using the `@accessible` and `@mockable` macros provided by
-[zio-macros][link-zio-macros] we get the _capability tags_, _access helper_ and _mockable implementation_ autogenerated for us:
+To reduce the amount of repetetive work ZIO provides a `@Mockable[A]` macro annotation which will generate _capability tags_ and _mock layer_ into annotated object:
 
 ```scala
-import zio.macros.annotation.{ accessible, mockable }
-import zio.console.Console
-
-@accessible(">")
-@mockable
-trait AccountObserver {
-  def accountObserver: AccountObserver.Service[Any]
-}
+type AccountObserver = Has[AccountObserver.Service]
 
 object AccountObserver {
-  trait Service[R] {
-    def processEvent(event: AccountEvent): ZIO[R, Nothing, Unit]
+  trait Service {
+    def processEvent(event: AccountEvent): ZIO[Any, Nothing, Unit]
   }
 
-  // autogenerated `object Service { ... }`
-  // autogenerated `object > extends Service[AccountObserver] { ... }`
-  // autogenerated `implicit val mockable: Mockable[AccountObserver] = ...`
+  def processEvent(event: AccountEvent) =
+    ZIO.accessM[AccountObserver](_.get.processEvent(event))
 }
+
+@Mockable[AccountObserver.Service]
```

Comment:
Any way to avoid having to specify the type parameter here?

Reply:
No. In previous approach we did not have to do that becouse we were annotating the companion object (in which the Service lied), however that was bad practice, as it was introducing `test` dependencies into your `src/main` code.

Here, we are generating the Mock machinery into a completely unrelated object and we need to somehow inform the macro for which service we are generating tags.